packages/superset-ui-chart-controls/test/utils/timeCompare.test.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { QueryObject, SqlaFormData } from '@superset-ui/core';
+import { timeCompareTransform } from '../../src';
+
+const formData: SqlaFormData = {
+  metrics: ['count(*)'],
+  time_range: '2015 : 2016',
+  granularity: 'month',
+  datasource: 'foo',
+  viz_type: 'table',
+};
+const queryObject: QueryObject = {
+  metrics: ['count(*)'],
+  time_range: '2015 : 2016',
+  granularity: 'month',
+  post_processing: [
+    {
+      operation: 'pivot',
+      options: {
+        index: ['__timestamp'],
+        columns: ['nation'],
+        aggregates: {
+          'count(*)': {
+            operator: 'sum',
+          },
+        },
+      },
+    },
+    {
+      operation: 'aggregation',
+      options: {
+        groupby: ['col1'],
+        aggregates: 'count',
+      },
+    },
+  ],
+};
+
+describe('timeCompareTransform', () => {
+  it('skip transformation', () => {
+    expect(timeCompareTransform(formData, queryObject).post_processing).toEqual(
```

Comment:
`timeCompareTransform` expects type of `QueryFormMetric[]` but `queryObject` is a single object of type `QueryObject`. These should probably be consistent (I think this is where the tests are failing)

Reply:
Hey, andrew. I have finished the refactoring. the UT all passed.